src/drivers/cmakeFileApiDriver.ts

```diff
@@ -259,13 +270,15 @@ export class CMakeFileApiDriver extends CMakeDriver {
         } else {
             log.debug(`Configuring using ${this.useCMakePresets ? 'preset' : 'kit'}`);
             log.debug('Invoking CMake', cmake, 'with arguments', JSON.stringify(args));
-            const env = await this.getConfigureEnvironment();
-            const res = await this.executeCommand(cmake, args, outputConsumer, { environment: env, cwd: this.binaryDir }).result;
+            const env = await this.getConfigureEnvironment(configurePreset);
+            const res = await this.executeCommand(cmake, args, outputConsumer, { environment: env, cwd: (configurePreset && configurePreset.binaryDir) ? configurePreset.binaryDir : this.binaryDir }).result;
```

Comment:
Line 239 already has this logic for picking the binaryDir. Consider storing it in a variable and using that instead.